capstone/src/main/resources/static/event.js

```diff
@@ -97,39 +97,42 @@ function selectedFilter(elementId){
  * @param userEmail current user's email
  */
 function createEventElement(eventListElement, event, isIndividual, userEvent, userEmail) {
-  const eventElement = createElement(eventListElement, 'li', '');
-  eventElement.className = 'event';
-
-  // Click for event detail modal
-  eventElement.addEventListener('click', () => {
-    showEventPage(event, isIndividual, userEmail);
-  });
+  fetch("get-public-profile?organization-id=" + event.organizationId).then(response => response.json()).then((data) => {
```

Comment:
hm.. I think the problem with printing in rank order comes from the varying return times from fetch

Reply:
Resolved